### PR TITLE
[develop] Update to virtualenv module removing pyvenv if Python greater than 3.6

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Beginning in Python 3.6, pyvenv has been deprecated in favor of using venv, so remove it from the list if we are using 3.6 or higher.

### What issues does this PR fix or reference?
N/A

### Tests written?
No.  Tests exist.

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

**Comment by gtmanfred** [comment]
Can you clarify this, is pyenv deprecated or pyvenv deprecated? Because they are two separate things.

File: salt/modules/virtualenv_mod.py [hunk 1]
```diff
@@ -139,9 +139,17 @@ def create(path,
         salt '*' virtualenv.create /path/to/new/virtualenv
     '''
     if venv_bin is None:
-        venv_bin = __pillar__.get('venv_bin') or __opts__.get('venv_bin')
-
-    cmd = [venv_bin]
+        # Beginning in 3.6, pyenv has been deprecated
```
Comment (gtmanfred) [inline salt/modules/virtualenv_mod.py]: this should be `pyvenv`

**Review by gtmanfred — COMMENTED** [review]
`pyenv` is a tool for installing multiple full versions of python, and is not part of the stdlib

`pyvenv` is a binary that used to be provided by python as a replacement for virtualenv, and in python3.6 was deprecated in favor of `python -m venv`

**Review by gtmanfred — APPROVED** [review]
Thanks

**Pull request merged**